builder/simplify.cc

```diff
@@ -391,25 +425,44 @@ class simplifier : public node_mutator {
       set_result(let_stmt::make(std::move(lets), std::move(body)));
     }
   }
+  
+  void merge_loop_references(symbol_map<symbol_info> add) {
```

Comment:
Add a parameter `bool is_in_loop = false`, and perhaps we should rename this back to just `merge_references` now that the loop-ness is optional.